### PR TITLE
prototype test for persistent tables

### PR DESCRIPTION
**Pull request by philip-stoev**
I wanted to make sure that the format of the test is acceptable before I add any more content.

I could not figure out a way to selectively pass or not pass ```--persistent-tables``` as an argument to the ```materialize``` service, so I had to create two separate services and so on. Please advise if this is the right way to do this.

I also had to implement ```kill-services``` . If there is a way to achieve the same outcome in the mzcompose.yml file without touching mzcompose.py, I will be glad to use it.

(The test currently fails due to a bug in table persistence)

### CODE REVIEW

**Comment by philip-stoev** [comment]
@ruchirK I meant to open this for Nikhil, but GitHub does not seem to be ready to handle 3-way parties like that. Will close this without merging as son as Nikhil has had the chance to look at it.

**Comment by ruchirK** [comment]
hey @philip-stoev I don't think Nikhil will be notified about this pull request as its on my personal fork of the repo? I'm not sure though. Did you ping him about this / is there any particular reason this wasn't on the main repo?

**Comment by philip-stoev** [comment]
Yes, I notified him manually in slack. This is not against the main repo because I needed --persistent-tables in order to write the test in first place. If I can not get the test reviewed this way, I am going to open a new PR after --persistent-tables lands in "main".

**Comment by ruchirK** [comment]
Ah got it! that all makes perfect sense

**Review by benesch — APPROVED** [review]
LGTM! Sorry for the delay.

File: misc/python/materialize/mzcompose.py [hunk 1]
```diff
@@ -533,6 +533,25 @@ def run(self, workflow: Workflow) -> None:
             raise errors.Failed(f"ERROR: services didn't come up cleanly: {services}")
 
 
+@Steps.register("kill-services")
+class RestartServicesStep(WorkflowStep):
```
[the remark this answers is not in the document]
Comment (philip-stoev) [reply misc/python/materialize/mzcompose.py:537]: Ops!  Copypasta. Thanks.

**Comment by philip-stoev** [comment]
Will be merged with a PR against the main Materialize repo.

**Pull request closed without merging**